src/com/amplitude/api/AmplitudeClient.java

```diff
@@ -284,6 +284,18 @@ public void run() {
                         lastIdentifyId = getLongvalue(LAST_IDENTIFY_ID_KEY, -1);
                         lastEventTime = getLongvalue(LAST_EVENT_TIME_KEY, -1);
 
+                        // install database reset listener to re-insert metadata in memory
+                        dbHelper.setDatabaseResetListener(new DatabaseResetListener() {
+                            @Override
+                            public void onDatabaseReset() {
+                                dbHelper.insertOrReplaceKeyValue(DEVICE_ID_KEY, deviceId);
+                                dbHelper.insertOrReplaceKeyValue(USER_ID_KEY, client.userId);
+                                dbHelper.insertOrReplaceKeyLongValue(OPT_OUT_KEY, optOut ? 1L : 0L);
+                                dbHelper.insertOrReplaceKeyLongValue(PREVIOUS_SESSION_ID_KEY, sessionId);
+                                dbHelper.insertOrReplaceKeyLongValue(LAST_EVENT_TIME_KEY, lastEventTime);
+                            }
```

Comment:
Should we be persisting everything that we have from getLongvalue()? (sequenceNumber, lastEventId, lastIdentifyId, lastEventTime)

Reply:
sequence number, last event id, last identify id correspond to the state of the event / identify tables in the database. on db reset those tables are dropped, so I think those counters should be cleared